### ISSUE TITLE
Grails 2.5.1 : Setting grails.project.fork.run to false causes crash

### ISSUE THREAD

**Issue by anthofo** [grails/grails-core#9164]
Hi,

Given the following config : 

```
grails.project.fork = [
    // configure settings for compilation JVM, note that if you alter the Groovy version forked compilation is required
    //  compile: [maxMemory: 256, minMemory: 64, debug: false, maxPerm: 256, daemon:true],

    // configure settings for the test-app JVM, uses the daemon by default
    test: [maxMemory: 768, minMemory: 64, debug: false, maxPerm: 256, daemon:true],
    // configure settings for the run-app JVM
    run: false,
    // configure settings for the run-war JVM
    war: [maxMemory: 768, minMemory: 64, debug: false, maxPerm: 256, forkReserve:false],
    // configure settings for the Console UI JVM
    console: [maxMemory: 768, minMemory: 64, debug: false, maxPerm: 256]
]
```

I get this error at startup : 

```
Error |
2015-08-03 16:54:00,055 [localhost-startStop-1] ERROR context.GrailsContextLoaderListener  - Error initializing the application: Error creating bean with name 'responseMimeTypesApi': Cannot resolve reference to bean 'mimeTypes' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'mimeTypes': FactoryBean threw exception on object creation; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jsonDataBindingSourceCreator': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private com.google.gson.Gson org.codehaus.groovy.grails.web.binding.bindingsource.JsonDataBindingSourceCreator.gson; nested exception is java.lang.NoSuchMethodError: org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.access$100(Lorg/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor;)Lorg/springframework/beans/factory/config/ConfigurableListableBeanFactory;
Message: Error creating bean with name 'responseMimeTypesApi': Cannot resolve reference to bean 'mimeTypes' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'mimeTypes': FactoryBean threw exception on object creation; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jsonDataBindingSourceCreator': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private com.google.gson.Gson org.codehaus.groovy.grails.web.binding.bindingsource.JsonDataBindingSourceCreator.gson; nested exception is java.lang.NoSuchMethodError: org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.access$100(Lorg/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor;)Lorg/springframework/beans/factory/config/ConfigurableListableBeanFactory;
    Line | Method
->>  266 | run       in java.util.concurrent.FutureTask
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|   1142 | runWorker in java.util.concurrent.ThreadPoolExecutor
|    617 | run . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run       in java.lang.Thread

Caused by BeanCreationException: Error creating bean with name 'mimeTypes': FactoryBean threw exception on object creation; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jsonDataBindingSourceCreator': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private com.google.gson.Gson org.codehaus.groovy.grails.web.binding.bindingsource.JsonDataBindingSourceCreator.gson; nested exception is java.lang.NoSuchMethodError: org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.access$100(Lorg/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor;)Lorg/springframework/beans/factory/config/ConfigurableListableBeanFactory;
->>  266 | run       in java.util.concurrent.FutureTask
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|   1142 | runWorker in java.util.concurrent.ThreadPoolExecutor
|    617 | run . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run       in java.lang.Thread

Caused by BeanCreationException: Error creating bean with name 'jsonDataBindingSourceCreator': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private com.google.gson.Gson org.codehaus.groovy.grails.web.binding.bindingsource.JsonDataBindingSourceCreator.gson; nested exception is java.lang.NoSuchMethodError: org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.access$100(Lorg/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor;)Lorg/springframework/beans/factory/config/ConfigurableListableBeanFactory;
->>  266 | run       in java.util.concurrent.FutureTask
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|   1142 | runWorker in java.util.concurrent.ThreadPoolExecutor
|    617 | run . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run       in java.lang.Thread

Caused by BeanCreationException: Could not autowire field: private com.google.gson.Gson org.codehaus.groovy.grails.web.binding.bindingsource.JsonDataBindingSourceCreator.gson; nested exception is java.lang.NoSuchMethodError: org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.access$100(Lorg/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor;)Lorg/springframework/beans/factory/config/ConfigurableListableBeanFactory;
->>  266 | run       in java.util.concurrent.FutureTask
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|   1142 | runWorker in java.util.concurrent.ThreadPoolExecutor
|    617 | run . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run       in java.lang.Thread

Caused by NoSuchMethodError: org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.access$100(Lorg/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor;)Lorg/springframework/beans/factory/config/ConfigurableListableBeanFactory;
->>  266 | run       in java.util.concurrent.FutureTask
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|   1142 | runWorker in java.util.concurrent.ThreadPoolExecutor
|    617 | run . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run       in java.lang.Thread
```

It works great with grails 2.4.* but not with 2.5.1

**Comment by noullet** [comment]
This issue was already present in Grails 2.5.0 (see https://jira.grails.org/browse/GRAILS-12129). Unfortunately it seems that it is not fixed in 2.5.1 despite the breaking change for several development teams.

It is now visible here: https://github.com/grails/grails-core/issues/662, where you will find some workarounds.

**Comment by graemerocher** [comment]
Closing as a duplicate of #662
